### PR TITLE
Add CleverReach Deploy API

### PR DESCRIPTION
**Pull request by jpbede**
This PR is adding the ability to deploy a certificate and key to a CleverReach account.

### CODE REVIEW

**Comment by jpbede** [comment]
Any news ? :)

File: deploy/cleverreach.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  _cca="$4"
+  _cfullchain="$5"
+
+  _debug _cdomain "$_cdomain"
+  _debug _ckey "$_ckey"
+  _debug _ccert "$_ccert"
+  _debug _cca "$_cca"
+  _debug _cfullchain "$_cfullchain"
+
+  _cleverreach_client_id="${DEPLOY_CLEVERREACH_CLIENT_ID}"
```
Comment (Neilpang) [inline deploy/cleverreach.sh:28]: please use `_getdeployconf `  and `_savedeployconf `

https://github.com/acmesh-official/acme.sh/blob/master/deploy/docker.sh#L22

File: deploy/cleverreach.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  _info "Obtaining a CleverReach access token"
+
+  _data="{\"grant_type\": \"client_credentials\", \"client_id\": \"${_cleverreach_client_id}\", \"client_secret\": \"${_cleverreach_client_secret}\"}"
+  _auth_result="$(_post "$_data" "https://rest.cleverreach.com/oauth/token.php" "" "POST" "application/json")"
+
+  _debug _data "$_data"
+  _debug _auth_result "$_auth_result"
+
+  _access_token=$(echo "$_auth_result" | _json_decode | jq -r .access_token)
```
Comment (Neilpang) [inline deploy/cleverreach.sh:51]: don't use `jq`

**Comment by jpbede** [comment]
Didn't expect that quick and a big thank you for the review :)

Have fixed your annotations.

**Comment by Neilpang** [comment]
did you add the usage here ?
https://github.com/acmesh-official/acme.sh/wiki/deployhooks

**Comment by jpbede** [comment]
No, i waited for the merge. I've added it now.

**Comment by Neilpang** [comment]
did you create  a new issue for bug reporting?

Example:  https://github.com/acmesh-official/acme.sh/issues/2977

**Comment by jpbede** [comment]
https://github.com/acmesh-official/acme.sh/issues/3276

**Pull request merged**